### DISCUSSION TITLE
Memory usage continuously increase under standalone mode

### DISCUSSION THREAD

**Discussion by JunlinPan-Crypto** [vercel/next.js#51812] [Help] [answered]
### Summary

I upgraded Nextjs from `v12.2` (everything is file)  to `v13.4.7` recently and found that the memory usage of nextjs service have been continuously increasing. The service is overloaded and can't respond the requests when the memory and cpu reach the resource limit.

My project is under `standalone` mode and running generated `node server.js` in AWS EKS cluster. 

I observed the memory usage gradually increase from 50MB to 400MB+ via PM2 monitor, and the CPU will reach 70%+ when it's not responding (service is still live at that time).

![WX20230626-152432@2x](https://github.com/vercel/next.js/assets/90383710/6393051d-a5aa-4509-8538-8860b1dfa119)

I used Chrome debugger to inspect the service memory. But I have no experience on diagnosing memory problem, spotting nothing helpful information.

**Comparison with initial heap snapshot (113MB vs 10MB):**

<img width="1514" alt="image" src="https://github.com/vercel/next.js/assets/90383710/bfbcefee-08f0-4718-9033-7a7e82bd2b02">

**Heap summary:**

<img width="1514" alt="image" src="https://github.com/vercel/next.js/assets/90383710/2ccdf3f9-e203-4413-a169-445a20456105">

**Allocation timeline:**

<img width="1514" alt="image" src="https://github.com/vercel/next.js/assets/90383710/9827410e-0627-4924-ae86-4c115d535809">

**Allocation sampling (Chart)**

<img width="1505" alt="image" src="https://github.com/vercel/next.js/assets/90383710/e0b69325-f6f9-4214-8a14-05365306baca">

**Allocation sampling (Heavy)**

<img width="1514" alt="image" src="https://github.com/vercel/next.js/assets/90383710/fc79593e-5e0a-4eee-b23e-96e2b2d153bd">

Are there any suggestions (like abnormal memory analyzing or downgrading, etc.) or related issues?

### Additional information

```js
`next info`:


Operating System:
      Platform: linux

Binaries:
      Node: 20.3.0
      npm: 9.6.7
      Yarn: N/A
      pnpm: 8.6.0
    Relevant packages:
      next: 13.4.7
      eslint-config-next: 13.4.6
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
```
```


### Example

_No response_

**Comment by JunlinPan-Crypto** [comment, marked as answer]
This issue may be related: #49929

**Comment by leerob** [comment]
Let's continue the conversation on that issue, please!
